summary for structure

I think it's good to add `summary` for structure too. parmed has this in interactive mode.

cpptraj has this

```python
In [53]: t.dihedral_info()
#       Mask [*] corresponds to 34 atoms.
#Dihedral    pk     phase pn                atoms
         1:  2.000 0.00  1.0 :1@O     :1@C     :2@N     :2@H     (12,11,13,14) O -C -N -H
E        2:  2.500 3.14  2.0 :1@O     :1@C     :2@N     :2@H     (12,11,13,14) O -C -N -H
         3:  0.000 0.00  2.0 :1@C     :2@N     :2@CA    :2@HA    (11,13,15,16) C -N -CX-H1
         4:  0.156 0.00  3.0 :1@HB3   :1@CB    :1@CA    :1@C     (10,7,5,11) HC-CT-CX-C
         5:  0.156 0.00  3.0 :1@HB2   :1@CB    :1@CA    :1@C     (9,7,5,11) HC-CT-CX-C
         6:  0.156 0.00  3.0 :1@HB1   :1@CB    :1@CA    :1@C     (8,7,5,11) HC-CT-CX-C
         7:  0.156 0.00  3.0 :1@HA    :1@CA    :1@CB    :1@HB1   (6,5,7,8) HP-CX-CT-HC
         8:  0.156 0.00  3.0 :1@HA    :1@CA    :1@CB    :1@HB2   (6,5,7,9) HP-CX-CT-HC
         9:  0.156 0.00  3.0 :1@HA    :1@CA    :1@CB    :1@HB3   (6,5,7,10) HP-CX-CT-HC
        10:  0.000 0.00  2.0 :1@HA    :1@CA    :1@C     :1@O     (6,5,11,12) HP-CX-C -O
```

parmed has this in interactive mode but not in normal (python) mode. 

Did I miss anything?

--- a/parmed/topologyobjects.py
+++ b/parmed/topologyobjects.py
@@ -3917,7 +3917,15 @@ class TrackedList(list):
         return list.__init__(self, *args)
 
     def __repr__(self):
-        return '%s(%s)' % (type(self).__name__, list.__repr__(self))
+        retstr = ['%s([\n' % (type(self).__name__)]
+        if len(self) > 30:
+            retstr.extend('\t%r\n' % self[i] for i in range(24))
+            retstr.append('\t...\n')
+            retstr.extend('\t%r\n' % self[i] for i in range(-5, 0))
+        else:
+            retstr.extend('\t%r\n' % i for i in self)
+        retstr.append('])')
+        return ''.join(retstr)
 
     @_changes
     def __delitem__(self, item):

